Warning when building mspan

I'm getting this warning about an extra semicolon when building mspan.
```
basix/mdspan.hpp:5185:57: error: extra ';' [-Werror=pedantic]
 5185 |     /* requires */ (extents_type::rank_dynamic()!=0)) {};
      |                                                         ^
```

When building with `-Werror=pedantic`, this causes build to fail

--- a/cpp/basix/mdspan.hpp
+++ b/cpp/basix/mdspan.hpp
@@ -5182,7 +5182,7 @@ public:
   MDSPAN_FUNCTION_REQUIRES(
     (MDSPAN_INLINE_FUNCTION_DEFAULTED constexpr),
     mdarray, (), ,
-    /* requires */ (extents_type::rank_dynamic()!=0)) {};
+    /* requires */ (extents_type::rank_dynamic()!=0)) {}
 #else
   MDSPAN_INLINE_FUNCTION_DEFAULTED constexpr mdarray() requires(extents_type::rank_dynamic()!=0) = default;
 #endif

